core/api.go

```diff
@@ -250,15 +252,21 @@ func (a *API) authnHandler(handler http.Handler) http.Handler {
 		req, err := auth.Authenticate(req)
 		if err != nil {
 			errorFormatter.Write(req.Context(), rw, errNotAuthenticated)
+			return
 		}
 		handler.ServeHTTP(rw, req)
 	})
 }
 
-func authzHandler(handler http.Handler) http.Handler {
+func (a *API) authzHandler(handler http.Handler) http.Handler {
+	auth := authz.NewAuthorizer(a.raftDB, grantPrefix, policyByRoute)
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		if !authz.Authorized(req.Context()) {
-			errorFormatter.Write(req.Context(), rw, errNotAuthorized)
+		err := auth.Authorize(req)
+		if err != nil {
+			// TODO(tessr/dominic): Switch the following to:
+			// errorFormatter.Write(req.Context(), rw, err)
+			// once dashboard can handle ErrNotAuthorized as well as errNotAuthenticated
+			errorFormatter.Write(req.Context(), rw, errNotAuthenticated)
```

Comment:
I think this makes some 500s look like 401s. Should this be

```go
if errors.Root(err) == authz.ErrNotAuthorized {
	err = errors.Sub(errNotAuthenticated, err)
}
if err != nil {
	errorFormatter.Write(req.Context(), rw, err)
}
```

And maybe that's not necessary if https://github.com/chain/chain/pull/943/files#r111655629 is fixed?